pkg/tagset/docs.go

```diff
@@ -6,20 +6,33 @@
 // Package tagset supports creation and manipulation of sets of tags.  It
 // does so in a safe and efficient fashion, supporting:
 //
-// - consistent hashing of tagsets to recognize commonalities
-// - flexible combination of tagsets from multiple sources
-// - immutability to allow re-use of tagsets
+//     - consistent hashing of tagsets to recognize commonalities
+//     - flexible combination of tagsets from multiple sources
+//     - immutability to allow re-use of tagsets
 //
 // The package otherwise presents a fairly abstract API that allows performance
 // optimizations without changing semantics.
 //
-// Accumulators
+// For background, see the proposal for this package at
+// https://github.com/DataDog/datadog-agent/blob/main/docs/proposals/agent-pkg-tagset-rfc.md
 //
-// HashlessTagsAccumulator and HashingTagsAccumulator both allow building tagsets bit-by-bit, by
-// appending new tags.
+// Tags
 //
-// HashedTags
+// The Tags type is an opaque, immutable data structure representing a set of
+// tags.  Agent code that handles tags, but does not manipulate them, need only
+// use this type.
 //
-// The HashedTags type represents an _immutable_ set of tags and associated hashes.
-// It is the primary data structure used to represent a set of tags.
+// Factories
+//
+// Factories are responsible for making new Tags instances.  Beneath a simple
+// interface, factories support optimization and deduplication.  A global factory
+// is available for general use, and purpose-specific factories can be created
+// for more intensive tag operations.
+//
+// In general, factories are not thread-safe
+//
+// Builders
+//
+// Builders are used to build tagsets tag-by-tag, before "freezing" into one or
+// more Tags instances.
```

Comment:
nit: Reading this with `godoc` it looks like the 'In general, factories are not thread-safe' is shown as a title.

![image](https://user-images.githubusercontent.com/5502710/144638519-80baaa74-93d6-41f0-9b4a-5142101cd13c.png)

Reply:
Ugh, I know the Go authors hate Markdown, but this is so much worse.

Reply:
Also, thanks for checking the docs!!!